docs/concepts.rst

```diff
@@ -15,12 +15,52 @@ An *event stream* is a time-ordered series of events.  For example, all the mous
 
 Conceptually, you can think of an event stream as an array of (time, value) pairs.  However, whereas an array is ordered by index, an event stream is ordered by its events' occurrence times.  You can't observe the second mouse click until after you've observed the first one.
 
+Event streams may be *infinite*, *finite*, or may *fail*.
+
+Infinite Event Streams
+^^^^^^^^^^^^^^^^^^^^^^
+
+An event stream may be infinite.  For example, :ref:`periodic` creates an infinite stream of events that occur at a regular period.  Limiting operations, such as :ref:`take` or :ref:`until`, are helpful in turning an infinite event stream into a finite one.
+
+Finite Event Streams
+^^^^^^^^^^^^^^^^^^^^
+
+An event stream may be finite.  A finite event stream will produce an *end signal* to indicate that it will never produce another event.  When an event stream ends, it will free underlying resources.
+
+.. _Failed Event Streams:
+
+Failed Event Streams
+^^^^^^^^^^^^^^^^^^^^
+
+An event stream may fail.  For example, when an event stream represents a resource, such as a websocket, and the resource fails or closes unexpectedly, the event stream *cannot* produce more events.  When an event stream fails, it will produce a *error signal* to indicate that it cannot produce more events.  The error signal includes a *reason* (an ``Error`` object) describing the failure.
+
+A failed event stream will attempt to free any underlying resources.
+
+The :ref:`recoverWith` operation allows you to handle an event stream failure.
+
+.. _Application Errors:
+
+Stream Failure vs. Application Errors
+`````````````````````````````````````
+
+Stream failures are different from *application errors*.  A stream failure indicates that an event stream *cannot* produce more events.  Application errors may or may not indicate an event stream failure.  For example, an event stream representing messages arriving on a websocket *fails* if the websocket is disconnected unexpectedly due to a wifi drop.  In contrast, an application error may occur in application logic that receives a websocket message and can't process because the message application state has changed.
```

Comment:
Change instances of "websocket" to WebSocket.

Maybe:
> In contrast, an application error may occur in the application logic, when a nonprocessible WebSocket message is received, because the message application state has changed.